### PR TITLE
feat: Add ExportedProgram as an IR

### PR DESCRIPTION
**Pull request by peri044**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/_compile.py [hunk 1]
```diff
@@ -91,8 +97,16 @@ def _get_target_ir(module_type: _ModuleType, ir: str) -> _IRType:
                     "Input graph is a Torchscript module but the ir provided is default (dynamo). Please set ir=torchscript to suppress the warning. Compiling the module with ir=torchscript",
                 )
                 return _IRType.ts
+            elif module_is_exportable:
+                raise ValueError(
+                    "Input graph is an ExportedProgram which is not currently supported. Please provide torch.nn.Module or torch.fx.GraphModule as input."
+                )
```
Comment (gs-olive) [inline py/torch_tensorrt/_compile.py:103]: Could add a note that support is coming soon, to differentiate this input type from completely unknown IR.

File: py/torch_tensorrt/_compile.py [hunk 2]
```diff
@@ -91,8 +97,16 @@ def _get_target_ir(module_type: _ModuleType, ir: str) -> _IRType:
...
             else:
                 raise ValueError("Module was provided in an unsupported format")
+        elif ir == "exported_program":
+            raise ValueError(
+                "ir=exported_program is not currently supported. Supported ir options : ts|fx|dynamo"
+            )
```
Comment (gs-olive) [inline py/torch_tensorrt/_compile.py:109]: Same as above for input type.

File: py/torch_tensorrt/_compile.py [hunk 3]
```diff
@@ -44,6 +46,7 @@ class _ModuleType(Enum):
     nn = 0
     ts = 1
     fx = 2
+    ep = 3
```
Comment (gs-olive) [inline py/torch_tensorrt/_compile.py:49]: Unrelated to this change, but the docstring on this `Enum` is not relevant to the content

File: py/torch_tensorrt/_compile.py [hunk 4]
```diff
@@ -36,6 +37,7 @@ class _IRType(Enum):
     fx = 1
     dynamo = 2
     torch_compile = 3
+    exported_program = 4
```
Comment (gs-olive) [inline py/torch_tensorrt/_compile.py:40]: Unrelated to this change, but the docstring on this `Enum` is not relevant to the content

**Review by gs-olive — COMMENTED** [review]
Overall, the addition looks good. Made comments on log messaging and docstrings.
